sbt-ash/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala

```diff
@@ -0,0 +1,65 @@
+package org.virtuslab.ash
+
+import sbt.Def
+import sbt.Keys._
+import sbt._
+
+object AkkaSerializationHelperPlugin extends AutoPlugin {
+
+  object autoImport extends AkkaSerializationHelperKeys {
+    val baseAshSettings: Seq[Def.Setting[_]] = AkkaSerializationHelperPlugin.baseAshSettings
+  }
+  import autoImport.{baseAshSettings => _, _}
+
+  override def projectSettings: Seq[Def.Setting[_]] = additionalSettings ++ ashSettings
+
+  override def globalSettings: Seq[Def.Setting[_]] = Nil
+
+  lazy val additionalSettings: Seq[Def.Setting[_]] = Seq(
+    libraryDependencies ++= Seq(
+        compilerPlugin(ashDumpEventSchemaCompilerPlugin.value),
+        compilerPlugin(ashRegistrationCheckerCompilerPlugin.value),
+        compilerPlugin(ashSerializabilityCheckerCompilerPlugin.value)),
+    scalacOptions ++= Seq(
+        s"-P:dump-event-schema-plugin:${target.value}",
+        s"-P:registration-checker-plugin:${target.value}"),
+    cleanFiles ++= Seq(
+        target.value / "dump-event-schema-cache",
+        target.value / "registration_checker_plugin_cache.csv"),
+    scalacOptions ++= {
+      val de = (ashDumpEventSchemaCompilerPlugin / ashCompilerPluginEnable).value
+      val dv = (ashDumpEventSchemaCompilerPlugin / ashCompilerPluginVerbose).value
+      val re = (ashRegistrationCheckerCompilerPlugin / ashCompilerPluginEnable).value
```

Comment:
Hmmm as I'm looking at that here... I think it'd be clearer if `registration checker` was renamed to `codec registration checker`, WDYT? can be done on a separate PR ofc